### PR TITLE
Transform tags

### LINKED ISSUE #84: Song/Album Tags need cleaning! - transform.py/load.py
Working on the SNS notifications has revealed an issue - many tags have trailing whitespaces and inconsistent casing. This needs to be dealt with in the transform script, with any other necessary changes made in the load script. This will ensure that the tag strings are easy to work with when implementing the topic subscriptions in SNS.

### PR DESCRIPTION
**Pull request by e-lemma**
Resolves #84 

- Tags are cleaned now
- Added tests for the cleaning function
- Amended test to reflect change in output

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -203,7 +203,8 @@ def insert_album_sale(cursor: DBCursor, album_sale: Dict[str, Any]) -> None:
         cursor, album_sale["item_description"], artist_id, album_sale["url"]
     )
     for tag in album_sale["album_tags"]:
-        get_or_insert_tags(cursor, tag, album_id)
+        if tag:
```
Comment (alina-101) [inline pipeline/load.py:206]: Nice error handling

File: pipeline/transform.py [hunk 2]
```diff
@@ -58,6 +58,13 @@ def clean_data(sales_data: dict) -> dict:
     }
 
 
+def clean_tags(tags: list[str]) -> list[str]:
+    """Removes leading/trailing whitespace and 'lowercases' a list of tags, returning it."""
+    if tags is None:
+        return []
+    return [tag.strip().lower() for tag in tags]
```
Comment (alina-101) [inline pipeline/transform.py:65]: nice normalisation. Some tags are proper nouns though. Should they be lowercase?

File: pipeline/test_transform.py [hunk 3]
```diff
@@ -140,10 +139,35 @@ def test_transform_sales_data_valid():
...
             "artist_url": "https://example_artist.bandcamp.com",
         }
     ]
 
     assert transform_sales_data(input_data) == expected_output
+
+
+def test_clean_tags_normal_case():
+    tags = [" TagOne ", "TAGTWO", "tagthree "]
+    expected = ["tagone", "tagtwo", "tagthree"]
```
Comment (alina-101) [inline pipeline/test_transform.py:154]: Nice range of tests for different cases

File: pipeline/transform.py (hunk 2 above)
Comment (e-lemma) [reply pipeline/transform.py:65]: Hmm...

Comment (e-lemma) [reply pipeline/transform.py:65]: It's a trade off between presentability on the dashboard and minimising duplication of tags, and in this case, it's worth the small loss in dashboard presentability.

**Review by alina-101 — APPROVED** [review]
Good reasoning for not implementing for my suggestion so I'm happy to approve

**Pull request merged**